webui/react/src/pages/TrialDetails/Profiles/ProfilesEnabled.tsx

```diff
@@ -1,66 +1,215 @@
+          acc.push(...filler);
+        }
+        return acc;
+      }, [ times || [], batches || [] ]);
+
+      // Pad the series config with blank series where applicable.
+      const series = metricKeys.reduce((acc, seriesKey) => {
+        const metricNames = metrics[seriesKey].names.slice(1);
+        if (seriesKey === key) {
+          const series = metricNames.map(seriesMapping);
+          acc.push(...series);
+        } else {
+          const filler = new Array(metricNames.length).fill(null).map(fillerMapping);
+          acc.push(...filler);
+        }
+        return acc;
+      }, [ timeSeries, batchSeries ]);
+
+      const options = { ...sharedOptions, axes: config[key].axes, series } as Options;
+
+      return { ...acc, [key]: { data, options } };
+    }, {} as Record<MetricType, { data: AlignedData, options: Options }>);
+
+    return uPlotData;
+  }, [ filters.name, metrics ]);
+
+  if (isLoading) {
+    return <Spinner spinning tip="Fetching system metrics..." />;
+  } else if (isEmpty) {
     return <Alert message="No data available." type="warning" />;
   }
 
   return (
     <>
-
       <Section
         bodyBorder
-        filters={<SystemMetricFilter />}
-        loading={systemMetrics.isLoading}
-        title="System Metrics"
-      >
-        <SystemMetricChart systemMetrics={systemMetrics} />
+        bodyNoPadding
+        loading={metrics[MetricType.Throughput].isLoading}
+        title="Throughput">
+        <UPlotChart
+          data={chartInfo[MetricType.Throughput].data}
+          options={chartInfo[MetricType.Throughput].options}
+          style={CHART_STYLE}
+        />
       </Section>
-
       <Section
-        bodyBorder={!timingMetrics.isEmpty}
-        loading={timingMetrics.isLoading}
-        title="Timing Metrics"
-      >
-        {timingMetrics.isEmpty
-          ? <Alert
+        bodyBorder={!metrics[MetricType.Timing].isEmpty}
+        bodyNoPadding
+        loading={metrics[MetricType.Timing].isLoading}
+        title="Timing Metrics">
+        {metrics[MetricType.Timing].isEmpty ? (
+          <Alert
             description="Timing metrics may not be available for your framework."
             message="No data found."
             type="warning"
           />
-          : <TimingMetricChart timingMetrics={timingMetrics} />
-        }
+        ) : (
+          <UPlotChart
+            data={chartInfo[MetricType.Timing].data}
+            options={chartInfo[MetricType.Timing].options}
+            style={CHART_STYLE}
+          />
+        )}
+      </Section>
+      <Section
+        bodyBorder
+        bodyNoPadding
+        filters={<SystemMetricFilter />}
+        loading={metrics[MetricType.System].isLoading}
+        title="System Metrics">
+        <UPlotChart
+          data={chartInfo[MetricType.System].data}
+          options={chartInfo[MetricType.System].options}
+          style={CHART_STYLE}
+        />
       </Section>
```

Comment:
bug: I don't know what's causing it, but when I switch system metrics, I am taken back up to the top of the page, which I didn't expect.

Reply:
oooo... this one is a doozie... it's not obvious but all 3 charts have an interdependency because they are now synced so when the filters update the system metrics, both throughput and timing metrics will have to re-render. So ultimately the entire section re-renders, where we lose the scroll position...

Reply:
ended up using a `useScroll` hook to remember the last scroll position and restoring it when chart changes.